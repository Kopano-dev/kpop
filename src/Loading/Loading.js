import React from 'react';
import PropTypes from 'prop-types';

import {FormattedMessage} from 'react-intl';

function Loading(props) {
  if (props.error) {
    // When the loader has errored.
    return <div id="loader">
      <FormattedMessage id="kpop.loader.error.message" defaultMessage="Error!"></FormattedMessage>
    </div>;
  } else if (props.timedOut) {
    // When the loader has taken longer than the timeout.
    return <div id="loader">
      <FormattedMessage id="kpop.loader.longtime.message" defaultMessage="Taking a long time..."></FormattedMessage>
    </div>;
  } else if (props.pastDelay) {
    // When the loader has taken longer than the delay.
    return <div id="loader">
      <FormattedMessage id="kpop.loader.loading.message" defaultMessage="Loading..."></FormattedMessage>
    </div>;
  } else {
    // When the loader has just started.
    return null;
  }
}

Loading.propTypes = {
  error: PropTypes.bool,
  timedOut: PropTypes.bool,
  pastDelay: PropTypes.bool,
};

export default Loading;
