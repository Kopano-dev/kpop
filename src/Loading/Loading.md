Loading error example:

```js
<Loading error={true}/>
```

Loading timeout example:

```js
<Loading timedOut={true}/>
```

Loading past delay example:

```js
<Loading pastDelay={true}/>
```
