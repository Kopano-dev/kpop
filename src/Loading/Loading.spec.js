import React from 'react';
import Loading from './Loading';

import createComponentWithIntl from '../utils/createComponentWithIntl';

test('Loading renders correctly', () => {
  const component = createComponentWithIntl(<Loading/>);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Loading renders correctly error=true', () => {
  const component =  createComponentWithIntl(<Loading error={true}/>);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Loading renders correctly timedOut=true', () => {
  const component = createComponentWithIntl(<Loading timedOut={true}/>);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Loading renders correctly pastDelay=true', () => {
  const component = createComponentWithIntl(<Loading pastDelay={true}/>);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
