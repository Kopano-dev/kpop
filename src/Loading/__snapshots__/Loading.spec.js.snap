// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders correctly 1`] = `null`;

exports[`Loading renders correctly error=true 1`] = `
<div
  id="loader"
>
  <span>
    Error!
  </span>
</div>
`;

exports[`Loading renders correctly pastDelay=true 1`] = `
<div
  id="loader"
>
  <span>
    Loading...
  </span>
</div>
`;

exports[`Loading renders correctly timedOut=true 1`] = `
<div
  id="loader"
>
  <span>
    Taking a long time...
  </span>
</div>
`;
